src/cast.c

```diff
@@ -126,35 +126,47 @@ r_obj* vec_cast_switch_native(const struct cast_opts* opts,
 }
 
 
-r_obj* vec_cast_default(r_obj* x,
-                        r_obj* to,
-                        r_obj* p_x_arg,
-                        r_obj* p_to_arg,
-                        struct r_lazy call,
-                        const struct fallback_opts* opts) {
+static inline
+r_obj* vec_cast_default_full(r_obj* x,
+                             r_obj* to,
+                             struct vctrs_arg* p_x_arg,
+                             struct vctrs_arg* p_to_arg,
+                             struct r_lazy call,
+                             const struct fallback_opts* opts,
+                             bool from_dispatch) {
   r_obj* df_fallback = KEEP(r_int(opts->df));
   r_obj* s3_fallback = KEEP(r_int(opts->s3));
+
+  r_obj* ffi_x_arg = KEEP(vctrs_arg(p_x_arg));
+  r_obj* ffi_to_arg = KEEP(vctrs_arg(p_to_arg));
+
   r_obj* ffi_call = KEEP(r_lazy_eval(call));
   r_obj* out = vctrs_eval_mask8(syms.vec_default_cast,
                                 syms_x, x,
                                 syms_to, to,
-                                syms_x_arg, p_x_arg,
-                                syms_to_arg, p_to_arg,
+                                syms_x_arg, ffi_x_arg,
+                                syms_to_arg, ffi_to_arg,
```

Comment:
Was this broken before?

Reply:
Oh no I see we were doing this earlier and oddly naming it with `r_obj* p_x_arg`